utils/network.go

```diff
@@ -62,123 +140,70 @@ func (n *Network) MarshalJSON() ([]byte, error) {
 }
 
 func (n *Network) Set(s string) error {
-	switch s {
-	case "MAINNET", "mainnet":
-		*n = Mainnet
-	case "GOERLI", "goerli":
-		*n = Goerli
-	case "GOERLI2", "goerli2":
-		*n = Goerli2
-	case "INTEGRATION", "integration":
-		*n = Integration
-	case "SEPOLIA", "sepolia":
-		*n = Sepolia
-	case "SEPOLIA_INTEGRATION", "sepolia-integration":
-		*n = SepoliaIntegration
-	default:
-		return ErrUnknownNetwork
+	predefinedNetworks := map[string]Network{
+		"MAINNET":             Mainnet,
+		"mainnet":             Mainnet,
+		"GOERLI":              Goerli,
+		"goerli":              Goerli,
+		"GOERLI2":             Goerli2,
+		"goerli2":             Goerli2,
+		"INTEGRATION":         Integration,
+		"integration":         Integration,
+		"SEPOLIA":             Sepolia,
+		"sepolia":             Sepolia,
+		"SEPOLIA-INTEGRATION": SepoliaIntegration,
+		"sepolia-integration": SepoliaIntegration,
 	}
-	return nil
-}
-
-func (n *Network) Type() string {
-	return "Network"
-}
 
-func (n *Network) UnmarshalText(text []byte) error {
-	return n.Set(string(text))
+	if network, ok := predefinedNetworks[s]; ok {
+		*n = network
+		return nil
+	}
+	return ErrUnknownNetwork
 }
 
-// baseURL returns the base URL without endpoint
-func (n Network) baseURL() string {
-	switch n {
-	case Goerli:
-		return "https://alpha4.starknet.io/"
-	case Mainnet:
-		return "https://alpha-mainnet.starknet.io/"
-	case Goerli2:
-		return "https://alpha4-2.starknet.io/"
-	case Integration:
-		return "https://external.integration.starknet.io/"
-	case Sepolia:
-		return "https://alpha-sepolia.starknet.io/"
-	case SepoliaIntegration:
-		return "https://integration-sepolia.starknet.io/"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
+func (n *Network) SetCustomNetwork(s string) error {
```

Comment:
I think we can remove this and just use `UnmarshalJSON` directly.

Reply:
Update: SetCustomNetwork is no longer needed so has been removed